brave/src/test/java/com/linecorp/armeria/common/brave/RequestContextCurrentTraceContextTest.java

```diff
@@ -17,71 +17,50 @@
+import org.mockito.junit.jupiter.MockitoExtension;
 
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
 import com.linecorp.armeria.common.RequestContext;
 import com.linecorp.armeria.common.util.SafeCloseable;
-import com.linecorp.armeria.internal.DefaultAttributeMap;
 import com.linecorp.armeria.internal.brave.TraceContextUtil;
 import com.linecorp.armeria.internal.brave.TraceContextUtil.PingPongExtra;
+import com.linecorp.armeria.server.ServiceRequestContext;
 
 import brave.propagation.CurrentTraceContext;
 import brave.propagation.CurrentTraceContext.Scope;
 import brave.propagation.TraceContext;
 import io.netty.channel.EventLoop;
-import io.netty.util.Attribute;
-import io.netty.util.AttributeKey;
 
-public class RequestContextCurrentTraceContextTest {
-    @Rule
-    public MockitoRule mockitoRule = MockitoJUnit.rule();
+@ExtendWith(MockitoExtension.class)
+class RequestContextCurrentTraceContextTest {
 
-    @Mock(answer = Answers.CALLS_REAL_METHODS)
-    RequestContext mockRequestContext;
-    @Mock(answer = Answers.CALLS_REAL_METHODS)
-    RequestContext mockRequestContext2;
+    @Nullable
+    RequestContext ctx;
     @Mock
     EventLoop eventLoop;
 
-    final CurrentTraceContext currentTraceContext = RequestContextCurrentTraceContext.ofDefault();
-    final DefaultAttributeMap attrs1 = new DefaultAttributeMap();
-    final DefaultAttributeMap attrs2 = new DefaultAttributeMap();
-    final TraceContext traceContext = TraceContext.newBuilder().traceId(1).spanId(1).build();
-
-    @Before
-    public void setup() {
-        when(mockRequestContext.eventLoop()).thenReturn(eventLoop);
-        when(mockRequestContext2.eventLoop()).thenReturn(eventLoop);
-        when(eventLoop.inEventLoop()).thenReturn(true);
-
-        when(mockRequestContext.attr(isA(AttributeKey.class))).thenAnswer(
-                (Answer<Attribute>) invocation -> attrs1.attr(invocation.getArgument(0)));
-        when(mockRequestContext2.attr(isA(AttributeKey.class))).thenAnswer(
-                (Answer<Attribute>) invocation -> attrs2.attr(invocation.getArgument(0)));
-    }
+    private static final CurrentTraceContext currentTraceContext =
+            RequestContextCurrentTraceContext.ofDefault();
+    private static final TraceContext traceContext = TraceContext.newBuilder().traceId(1).spanId(1).build();
 
-    @Test
-    public void copy() {
-        try (SafeCloseable requestContextScope = mockRequestContext.push()) {
-            try (Scope traceContextScope = currentTraceContext.newScope(traceContext)) {
-                TraceContextUtil.copy(mockRequestContext, mockRequestContext2);
-                assertThat(attrs1.attrs().next().get())
-                        .isEqualTo(traceContext)
-                        .isEqualTo(attrs2.attrs().next().get());
-            }
-        }
+    @BeforeEach
+    void setup() {
+        Mockito.lenient().when(eventLoop.inEventLoop()).thenReturn(true);
```

Comment:
Better to add `@MockitoSettings` to the class to say to use lenient stubs

Reply:
Added!